# CLinical

In [55]:
import pandas as pd 

df_clinical = pd.read_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/clinical.csv")
df_clinical.insert(1, 'slide_id', 'x')
clinical_patients = set(df_clinical["case_id"].unique())
len(clinical_patients), df_clinical.head()

(1099,
         case_id slide_id  type   age  gender  censorship    survival  race
 0  TCGA-CS-4938        x     0  31.0       1           0  117.500082     0
 1  TCGA-CS-4941        x     0  67.0       0           1    7.693066     0
 2  TCGA-CS-4942        x     0  44.0       1           1   43.889930     1
 3  TCGA-CS-4943        x     0  37.0       0           1   36.361245     0
 4  TCGA-CS-4944        x     0  50.0       0           0   60.097972     0)

# CNA & RNA

In [20]:
import pandas as pd

cna = pd.read_csv("combined_CNA_filtered.csv")
rna = pd.read_csv("combined_RNA_filtered.csv")
mut = pd.read_csv("combined_MUT_filtered.csv")
clinical = pd.read_csv("clinical.csv")

In [48]:
cna.shape, rna.shape, clinical.shape

((1086, 2924), (674, 2856), (1099, 7))

In [ ]:
cna_clinical = pd.merge(clinical,cna, left_on="case_id", right_on="PatientID", how="inner")
# Custom Dataset Expects slide id of pathology image in a column... for now we will add it as a dummy entry untill we do the pathology
cna_clinical.insert(
    loc=1,                # index position (0 = first col, 1 = after first col)
    column="slide_id",       # new column name
    value="x"              # same dummy value for all rows
)

cna_clinical.isna().sum().sort_values(ascending=False)


age           283
case_id         0
PRKCQ_cna       0
PRKACB_cna      0
PRKACG_cna      0
             ... 
HES1_cna        0
HES2_cna        0
HES4_cna        0
HES5_cna        0
ZXDC_cna        0
Length: 2932, dtype: int64

In [50]:
cna_clinical = cna_clinical.dropna(subset=['survival'])
cna_clinical.to_csv("cna_clinical.csv")
cna_clinical.shape, cna_clinical.head()


((1081, 2932),
         case_id slide_id  type   age  gender  censorship    survival  race  \
 0  TCGA-CS-4938        x     0  31.0       1           0  117.500082     0   
 1  TCGA-CS-4941        x     0  67.0       0           1    7.693066     0   
 2  TCGA-CS-4942        x     0  44.0       1           1   43.889930     1   
 3  TCGA-CS-4943        x     0  37.0       0           1   36.361245     0   
 4  TCGA-CS-4944        x     0  50.0       0           0   60.097972     0   
 
       PatientID  AAK1_cna  ...  ZSCAN12_cna  ZSCAN2_cna  ZSCAN20_cna  \
 0  TCGA-CS-4938         0  ...            0           0            0   
 1  TCGA-CS-4941         0  ...            0           0            1   
 2  TCGA-CS-4942         0  ...            1           0            0   
 3  TCGA-CS-4943         0  ...            0           0            0   
 4  TCGA-CS-4944         0  ...            0           0            0   
 
    ZSCAN21_cna  ZSCAN22_cna  ZSCAN26_cna  ZSCAN9_cna  ZXDA_cna  ZXDB

In [33]:
rna_clinical = pd.merge(clinical,rna, left_on="case_id", right_on="PatientID", how="inner")
# Custom Dataset Expects slide id of pathology image in a column... for now we will add it as a dummy entry untill we do the pathology
rna_clinical.insert(
    loc=1,                # index position (0 = first col, 1 = after first col)
    column="slide_id",       # new column name
    value="x"              # same dummy value for all rows
)
rna_clinical.isna().sum().sort_values(ascending=False)
# rna_clinical.to_csv("rna_clinical.csv")
# rna_clinical.shape, rna_clinical.head()

PPBPP1_rna     674
PRLH_rna       674
TSSK2_rna      674
HSFX1_rna      674
TGIF2LY_rna    160
              ... 
HERPUD1_rna      0
HES1_rna         0
HES2_rna         0
HES4_rna         0
ZXDC_rna         0
Length: 2864, dtype: int64

In [35]:
missing = rna_clinical.isna().sum()
missing[missing>0]

age              54
survival          2
CDX4_rna        160
CSH1_rna        160
CSHL1_rna       160
DEFB104A_rna    160
FGF4_rna        160
FGF6_rna        160
GH2_rna         160
HSFX1_rna       674
HTN3_rna        160
IFNA16_rna      160
IFNA7_rna       160
IL1F10_rna      160
INS_rna         160
LACRT_rna       160
LEUTX_rna       160
PPBPP1_rna      674
PRLH_rna        674
RETNLB_rna      160
SSX7_rna        160
SSX9P_rna       160
TGIF2LX_rna     160
TGIF2LY_rna     160
TSSK2_rna       674
dtype: int64

In [46]:
rna_clinical = rna_clinical.dropna(subset=['survival'])
protected = {'case_id', 'label', 'survival', 'censorship', 'age', 'gender'}
df = rna_clinical.drop(columns=[c for c in rna_clinical.columns if c not in {'age', 'survival'} and rna_clinical[c].isna().any()], inplace=True)

rna_clinical.to_csv("rna_clinical.csv")

missing = rna_clinical.isna().sum()
missing[missing>0]

age    52
dtype: int64

In [20]:
#Number of patients who have both cna and rna
print(len(set(cna_clinical.case_id.to_list())))
print(len(set(rna_clinical.case_id.to_list())))
len((set(cna_clinical.case_id.to_list())).intersection(set(rna_clinical.case_id.to_list())))

1086
669


661

In [22]:
cna_clinical

,case_id,slide_id,type,age,gender,censorship,survival,race,PatientID,AAK1_cna,...,ZSCAN12_cna,ZSCAN2_cna,ZSCAN20_cna,ZSCAN21_cna,ZSCAN22_cna,ZSCAN26_cna,ZSCAN9_cna,ZXDA_cna,ZXDB_cna,ZXDC_cna
0,TCGA-CS-4938,x,0,31.0,1,0,117.500082,0,TCGA-CS-4938,0,...,0,0,0,1,0,0,0,1,1,0
1,TCGA-CS-4941,x,0,67.0,0,1,7.693066,0,TCGA-CS-4941,0,...,0,0,1,1,0,0,0,0,0,0
2,TCGA-CS-4942,x,0,44.0,1,1,43.889930,1,TCGA-CS-4942,0,...,1,0,0,1,-1,1,1,0,0,0
3,TCGA-CS-4943,x,0,37.0,0,1,36.361245,0,TCGA-CS-4943,0,...,0,0,0,0,-1,0,0,0,0,0
4,TCGA-CS-4944,x,0,50.0,0,0,60.097972,0,TCGA-CS-4944,0,...,0,0,0,0,-1,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1081,TCGA-02-0111,x,1,NaN,0,1,23.177828,1,TCGA-02-0111,0,...,0,0,0,1,1,0,0,0,0,0
1082,TCGA-19-0955,x,1,NaN,0,0,11.769734,1,TCGA-19-0955,0,...,0,0,0,1,1,0,0,0,0,0
1083,TCGA-14-0781,x,1,NaN,0,1,0.953414,1,TCGA-14-0781,0,...,0,0,0,1,0,0,0,0,0,0
1084,TCGA-06-0412,x,1,NaN,0,1,9.567018,1,TCGA-06-0412,0,...,0,0,0,1,0,0,0,0,0,0


In [27]:
missing = cna_clinical.isna().any()


In [30]:
cna_clinical.isna().sum().sort_values(ascending=False)


age           288
survival        5
case_id         0
PRKCQ_cna       0
PRKACB_cna      0
             ... 
HES2_cna        0
HES4_cna        0
HES5_cna        0
HES6_cna        0
ZXDC_cna        0
Length: 2932, dtype: int64

# MUT

In [58]:
import pandas as pd
df_mut = pd.read_csv('/Users/alaa.mohamed/Desktop/Personal/Projects/DeepMultiModalFusion/data/mut_full_df.csv')


In [59]:
df_mut = df_mut.drop(columns = "Unnamed: 0")
df_mut = df_mut.drop_duplicates("case_id")
len(df_mut.case_id.unique())
df_mut["slide_id"] = "x"

In [60]:
mut_patients = set(df_mut.case_id.unique())
len(mut_patients), len(mut_patients.intersection(clinical_patients))


(895, 895)

In [67]:
# df_mut = df_mut.drop(columns = "index")
df_mut

,case_id,NR4A2_mut,ZBTB40_mut,IRF9_mut,HIF3A_mut,IKZF1_mut,PPARG_mut,BRAF_mut,CD44_mut,STK32B_mut,...,LILRA6_mut,TBX15_mut,PRKCQ_mut,AIRE_mut,HNF4A_mut,TNFRSF11B_mut,ANKK1_mut,BTAF1_mut,MAPK7_mut,SMAD9_mut
0,TCGA-02-0003,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
1,TCGA-02-0033,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
2,TCGA-02-0047,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3,TCGA-02-0055,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
4,TCGA-02-2466,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
890,TCGA-WY-A85A,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
891,TCGA-WY-A85B,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
892,TCGA-WY-A85C,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
893,TCGA-WY-A85D,0,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [68]:
df_mut_full = pd.merge(df_clinical, df_mut, on="case_id", how="inner")
df_mut_full.shape

(895, 1445)

In [71]:
df_mut_full.to_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/mut_895_patients.csv")
df_mut_full

,case_id,slide_id,type,age,gender,censorship,survival,race,NR4A2_mut,ZBTB40_mut,...,LILRA6_mut,TBX15_mut,PRKCQ_mut,AIRE_mut,HNF4A_mut,TNFRSF11B_mut,ANKK1_mut,BTAF1_mut,MAPK7_mut,SMAD9_mut
0,TCGA-CS-4938,x,0,31.0,1,0,117.500082,0,0,0,...,0,0,0,0,0,0,0,0,0,0
1,TCGA-CS-4941,x,0,67.0,0,1,7.693066,0,0,0,...,0,0,0,0,0,0,0,0,0,0
2,TCGA-CS-4942,x,0,44.0,1,1,43.889930,1,0,0,...,0,0,0,0,0,0,0,0,0,0
3,TCGA-CS-4943,x,0,37.0,0,1,36.361245,0,0,0,...,0,0,0,0,0,0,0,0,0,0
4,TCGA-CS-4944,x,0,50.0,0,0,60.097972,0,0,0,...,0,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
890,TCGA-28-2501,x,1,NaN,0,1,NaN,1,0,0,...,0,0,0,0,0,0,0,0,0,0
891,TCGA-06-0214,x,1,NaN,0,1,15.024493,1,0,0,...,0,0,0,0,0,0,0,0,0,0
892,TCGA-06-0139,x,1,NaN,0,1,11.901239,1,0,0,...,0,0,0,0,0,0,0,0,0,0
893,TCGA-16-1048,x,1,NaN,0,1,NaN,1,0,0,...,0,0,0,0,0,0,0,0,0,0


In [72]:

mut_patients = set(df_mut.case_id.unique())
len(mut_patients)

895

# CNA

In [73]:
df_cna= pd.read_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/cna_232df.csv")
df_cna.head()

,Unnamed: 0.2,Unnamed: 0,Unnamed: 0.1,case_id,slide_id,type,age,gender,survival,censorship,...,ZSCAN12_cna,ZSCAN2_cna,ZSCAN20_cna,ZSCAN21_cna,ZSCAN22_cna,ZSCAN26_cna,ZSCAN9_cna,ZXDA_cna,ZXDB_cna,ZXDC_cna
0,0,0,0,TCGA-06-0128,TCGA-06-0128-01Z-00-DX3.37955b76-3fbf-41c5-ba5...,1,66.0,1.0,22.7,0.0,...,0,-1,0,1,0,0,0,-1,-1,-1
1,1,1,1,TCGA-06-0213,TCGA-06-0213-01Z-00-DX5.09126b9a-a487-4a3e-ad6...,1,55.0,0.0,0.5,0.0,...,0,-1,1,1,0,0,0,0,0,0
2,2,2,2,TCGA-12-0616,TCGA-12-0616-01Z-00-DX4.bcc9933d-865c-4c8d-a96...,1,36.0,0.0,14.7,0.0,...,-1,0,0,2,0,-1,-1,0,0,0
3,3,3,3,TCGA-06-5412,TCGA-06-5412-01Z-00-DX1.039632bd-2493-4a78-839...,1,78.0,0.0,4.5,0.0,...,0,0,0,1,0,0,0,0,0,0
4,4,4,4,TCGA-HT-7690,TCGA-HT-7690-01Z-00-DX2.70B38C74-164C-466F-BE3...,0,29.0,1.0,0.1,1.0,...,0,0,0,0,0,0,0,-1,-1,0


In [76]:
# df_cna = df_cna.drop(columns =[ "Unnamed: 0", "Unnamed: 0.1", "Unnamed: 0.2"])

df_cna = df_cna.drop_duplicates("case_id")
df_cna["slide_id"] = "x"
df_cna.reset_index(inplace=True)
df_cna = df_cna.drop(columns =[ "index"])

df_cna.shape

/var/folders/zd/f_dgjh_56ll03xfpgr7q1qtr0000gp/T/ipykernel_31331/1016034082.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_cna["slide_id"] = "x"


(232, 2930)

In [77]:
cna_patients = set(df_cna.case_id.unique())
len(cna_patients)

232

In [84]:
patients_with_cna_and_mut = mut_patients.intersection(cna_patients)
len(patients_with_cna_and_mut)

177

In [80]:
df_cna.columns, df_mut_full.columns

(Index(['case_id', 'slide_id', 'type', 'age', 'gender', 'survival',
        'censorship', 'AAK1_cna', 'AATF_cna', 'AATK_cna',
        ...
        'ZSCAN12_cna', 'ZSCAN2_cna', 'ZSCAN20_cna', 'ZSCAN21_cna',
        'ZSCAN22_cna', 'ZSCAN26_cna', 'ZSCAN9_cna', 'ZXDA_cna', 'ZXDB_cna',
        'ZXDC_cna'],
       dtype='object', length=2930),
 Index(['case_id', 'slide_id', 'type', 'age', 'gender', 'censorship',
        'survival', 'race', 'NR4A2_mut', 'ZBTB40_mut',
        ...
        'LILRA6_mut', 'TBX15_mut', 'PRKCQ_mut', 'AIRE_mut', 'HNF4A_mut',
        'TNFRSF11B_mut', 'ANKK1_mut', 'BTAF1_mut', 'MAPK7_mut', 'SMAD9_mut'],
       dtype='object', length=1445))

In [81]:
# Identify overlapping columns (excluding the merge key)
overlap_cols = set(df_cna.columns) & set(df_mut_full.columns) - {'case_id'}
print(f"Overlapping columns: {overlap_cols}")

# Keep only unique columns from df_mut 
mut_unique_cols = [col for col in df_mut_full.columns if col not in overlap_cols or col == 'case_id']
df_mut_clean = df_mut_full[mut_unique_cols]

# Merge
merged_df = pd.merge(df_cna, df_mut_clean, on='case_id', how='inner')
merged_df.head()

Overlapping columns: {'age', 'censorship', 'gender', 'type', 'survival', 'slide_id'}


,case_id,slide_id,type,age,gender,survival,censorship,AAK1_cna,AATF_cna,AATK_cna,...,LILRA6_mut,TBX15_mut,PRKCQ_mut,AIRE_mut,HNF4A_mut,TNFRSF11B_mut,ANKK1_mut,BTAF1_mut,MAPK7_mut,SMAD9_mut
0,TCGA-06-0128,x,1,66.0,1.0,22.7,0.0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
1,TCGA-06-0213,x,1,55.0,0.0,0.5,0.0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
2,TCGA-12-0616,x,1,36.0,0.0,14.7,0.0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3,TCGA-06-5412,x,1,78.0,0.0,4.5,0.0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
4,TCGA-HT-7690,x,0,29.0,1.0,0.1,1.0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [83]:
merged_df.shape
merged_df.to_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/mut_cna_177_patients.csv")

for comparison and analysis of results we will filter to have a copy of df with cna data alone for the 177 patients and same for mut

In [88]:
df_cna_177 = df_cna[df_cna["case_id"].isin(patients_with_cna_and_mut)]
df_cna_177.to_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/cna_177_patients.csv")
df_cna_177.shape

(177, 2930)

In [87]:
df_mut_177 = df_mut_full[df_mut_full["case_id"].isin(patients_with_cna_and_mut)]
df_mut_177.to_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/mut_177_patients.csv")
df_mut_177.shape

(177, 1445)

# PyRadiomics 1D

In [ ]:
import pandas as pd

gbm_df = pd.read_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/1D_MRI/GBM_pyradiomic_features.csv")
lgg_df = pd.read_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/1D_MRI/LGG_pyradiomic_features.csv")

In [ ]:
# Add tumor type labels
gbm_df['Tumor_Type'] = 'GBM'
lgg_df['Tumor_Type'] = 'LGG'

print(f"✓ GBM samples: {len(gbm_df)}")
print(f"✓ LGG samples: {len(lgg_df)}")

# Merge radiomics datasets
radiomics_df = pd.concat([gbm_df, lgg_df], ignore_index=True)
print(f"✓ Combined radiomics samples: {len(radiomics_df)}")

# Load clinical data
print("Loading clinical data...")
clinical_df = pd.read_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/clinical.csv")
print(f"✓ Clinical data samples: {len(clinical_df.case_id.unique())}")

# Merge with clinical data
merged_df = pd.merge(clinical_df, radiomics_df, left_on="case_id", right_on="PatientID" ,how='inner')

print(f"✓ Final merged dataset: {len(merged_df)} samples")
print(f"✓ Total features: {len(merged_df.columns)}")

# Display basic info
print("\nDataset Info:")
print(f"- Shape: {merged_df.shape}")
print(f"- Tumor type distribution:")
print(merged_df['Tumor_Type'].value_counts())

# Save merged dataset
merged_df.to_csv("/Users/alaa.mohamed/Desktop/MMContraFuse/data/processed_tabular_data/radio1D_clinical.csv", index=False)


✓ GBM samples: 102
✓ LGG samples: 65
✓ Combined radiomics samples: 167
Loading clinical data...
✓ Clinical data samples: 1099
✓ Final merged dataset: 163 samples
✓ Total features: 457

Dataset Info:
- Shape: (163, 457)
- Tumor type distribution:
Tumor_Type
GBM    98
LGG    65
Name: count, dtype: int64
